pkg/kgo/client.go

```diff
@@ -215,6 +201,50 @@ func NewClient(opts ...Opt) (*Client, error) {
 	return cl, nil
 }
 
+// Parse broker IP/host and port from a string, using the default Kafka port if
+// unspecified. Supported address formats:
+//
+// - IPv4 host/IP without port: "127.0.0.1", "localhost"
+// - IPv4 host/IP with port: "127.0.0.1:1234", "localhost:1234"
+// - IPv6 IP without port:  "[2001:1000:2000::1]", "::1"
+// - IPv6 IP with port: "[2001:1000:2000::1]:1234"
+func parseBrokerAddr(addr string) (hostport, error) {
+	// Bracketed IPv6
+	if strings.IndexByte(addr, '[') == 0 {
```

Comment:
I've always written this as `len(addr) > 0 && addr[0] == '['` but this works too